File: JavaProjectFE/jewelry-shop/src/app/models/product.ts
import { SafeUrl } from "@angular/platform-browser";

export interface ProductImage {
    imageId: string;
    productId: string;
    imageUrl: string;
    altText: string;
  }
  
export interface ProductResponse {
    productId:string
    name: string;
    description: string;
    price: number;
    stockQuantity: number;
    categoryName: string;
    brandName: string;
    productImages: ProductImage[];
    imageUrl?: SafeUrl;
    materials : Material[];
  }
export interface ProductRequest {
  name: string;
  description: string;
  price: number;
  originalPrice: number;
  stockQuantity: number;
  categoryId: string;
  brandId: string;
}

export interface Material {
  name: string;
  description: string;
}

File: JavaProjectBE/product-service/src/main/java/org/jewelryshop/productservice/entities/Product.java
package org.jewelryshop.productservice.entities;

import lombok.*;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class Product {
    private String productId;
    private String name;
    private String description;
    private double price;
    private double originalPrice;
    private int stockQuantity;
    private String categoryId;
    private String brandId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private List<ProductImage> productImages;
    private Set<Material> materials;
}
